File: izinga-ordermanager/src/main/java/io/curiousoft/izinga/ordermanagement/notification/EmailRequest.java
package io.curiousoft.izinga.ordermanagement.notification;

import io.curiousoft.izinga.commons.model.BasketItem;
import io.curiousoft.izinga.commons.model.Order;
import io.curiousoft.izinga.commons.model.StoreProfile;
import io.curiousoft.izinga.commons.model.UserProfile;

import java.util.ArrayList;
import java.util.List;

class Data{
    public String account_name;
    public StoreProfile store;
    public UserProfile customer;
    public Order order;
    public List<BasketItem> items;
}

class From{
    public String email;
}

class Personalization {
    public String email;
    public Data data;

    public Personalization(String email, Data data) {
        this.email = email;
        this.data = data;
    }
}

public class EmailRequest{
    public From from;
    public ArrayList<To> to;
    public ArrayList<Personalization> personalization;
    public String template_id;
}

class To {
    public String email;
    public To(String email) {
        this.email = email;
    }
}

File: izinga-recon/src/main/kotlin/io/curiousoft/izinga/recon/notification/EmailRequest.kt
package io.curiousoft.izinga.recon.notification

import io.curiousoft.izinga.commons.model.Order
import io.curiousoft.izinga.recon.payout.Payout
import java.text.SimpleDateFormat

internal class Data(val payout: Payout) {
    val order: List<OrderReport> = payout.orders.map { OrderReport(it) }
    val date: String = SimpleDateFormat("YYYY/MM/dd HH:mm").format(payout.createdDate)

}

internal class OrderReport(order: Order) {
    var date: String? = null
    var id: String? = null
    var basketAmount: String? = null

    init {
        this.id = order.id
        this.date = SimpleDateFormat("YYYY/MM/dd HH:mm").format(order.createdDate)
        this.basketAmount = "R${order.basketAmount}"
    }
}

internal class From {
    var email: String? = null
}

internal class Personalization(var email: String, var data: Data)

internal class EmailRequest (
    var from: From? = null,
    var to: List<To>? = null,
    var personalization: List<Personalization>? = null,
    var template_id: String? = null
)

internal class To(var email: String)